### ISSUE TITLE
Keyboard arrows move the cell selection even when a custom editor is focused

### ISSUE THREAD

**Issue by bmakan** [adazzle/react-data-grid#1696]
- Have you already searched for similar issues? Please help us out and double-check first!

- Also, please don't be that person who deletes this template. It's here for a reason.

- Thanks!

---

### Which version of React JS are you using?

✅ Officially supported ✅
- [ ] v15.4.x

⚠️ Not officially supported, expect warnings ⚠️
- [ ] v15.5.x
- [ ] v15.6.x

☣️ Not officially supported, expect warnings and errors ☣️
- [x] v16.x.x

---

### Which browser are you using?

✅ Officially supported ✅
- [ ] IE 9 / IE 10 / IE 11
- [ ] Edge
- [x] Chrome

⚠️ Not officially supported, but "should work" ⚠️
- [x] Firefox
- [ ] Safari

---

### I'm submitting a ...

- [x] 🐛 Bug Report
- [ ] 💡 Feature Request

> 👋 Need general support? Not sure about how to use React itself, or how to get started with the Grid?
> Please do not submit support request here. Instead see

> https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md

---

### Issue Details

```
Please include:
- What the current behavior is
- What the desired behvaior is
- (If Bug) Steps to reproduce the issue
- (If Feature) The motivation / use case for the feature

We especially love screenshots / videos of problems, and remember
The Best Issue Is A Pull Request™
```

When a custom editor is open, keyboard navigation keys will move the grid selection and therefore render new editor for newly selected cell.

Seems like the `EditorContainer` event's `stopPropagation` fails for some reason.

As a workaround, I provided my own `event.stopPropagation` on keyboard navigation keys.

**Comment by pixel9** [comment]
This issue is similar to this one: https://github.com/adazzle/react-data-grid/issues/1559

From what I can tell, the [`EditorContainer`](https://github.com/adazzle/react-data-grid/blob/d27999498f3e94a0d8c714825411860864949ea4/packages/react-data-grid/src/common/editors/EditorContainer.tsx) component was designed for in-cell type editors that are effectively a _single_ form input element with additional logic.

I'm running into this behavior with a fairly complex editor that opens in a modal (similar to how the color picker works in this [example](https://adazzle.github.io/react-data-grid/docs/examples/custom-editors). Even the color picker example in the documentation exhibits similar issues where tab navigation exits the entire picker instead of advancing to the next input inside of the picker.

Is there a hook that can disable all keyboard navigation on the grid while a complex editor is open? If not, what should the API for that look like? I'm willing to try and fix this but I don't want to submit a solution that goes completely against/breaks the intended design.

**Comment by antrix1** [comment]
I was building a custom remote autocomplete editor when I realized I couldn't navigate the dropdown menu with arrow keys.

For my use case, I was able to customize the behavior because [onPressArrowUpOrDown](https://github.com/adazzle/react-data-grid/blob/8f02fac3a0e7a8a53852273546fc31c57365f72f/packages/react-data-grid/src/common/editors/EditorContainer.tsx#L146-L148) handler has a hook to stop event propagation in case the editor has some results displayed.

I just implemented `hasResults` method in my editor component to return true when the dropdown is active, e.g.: `hasResults = () => this.state.suggestions.length > 0;`

There's currently not a standardized way to disable/hook into each of the events, so I guess the easiest way is to wrap a custom editor with a container element and just `stopPropagation` on its `keyDown` event of interest.
